### ISSUE TITLE
TableSelection: unnecessary horizontal overflow 

### ISSUE THREAD

**Issue by Lukas742** [SAP/ui5-webcomponents#9679]
### Bug Description

If the `Table` is implementing the selection feature, there is an unnecessary horizontal overflow for some sizes. Also, some header texts are cut-off. 

<img width="964" alt="image" src="https://github.com/user-attachments/assets/a54930aa-08a2-418e-8c7e-a7e4b16a94a2">

### Affected Component

ui5-table, ui5-table-selection

### Expected Behaviour

If enough space is available, there shouldn't be an overflow (no scrollbar) and the header text should always be visible (if not restricted by e.g. column width).

### Isolated Example

https://stackblitz.com/edit/github-ltccua-sqsrrf?file=main.js,index.html,package.json

### Steps to Reproduce

1. Go to [StackBlitz](https://stackblitz.com/edit/github-ltccua-sqsrrf?file=main.js,index.html,package.json)
2. See that although there'd be enough space for all the content to be visible, there's an scrollbar displayed and some of the headers are cut-off.
3. You can observe the same in your [playground](https://sap.github.io/ui5-webcomponents/components/TableSelection/)

### Priority

None

### UI5 Web Components Version

2.1.1

### Browser

Chrome

### Organization

UI5WCR

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by unazko** [comment]
Hello @SAP/ui5-webcomponents-topic-core,

The issue is reproducible via the provided by @Lukas742 example.
Please do follow up on this issue.

Best regards,
Boyan

**Comment by ilhan007** [comment]
Hello @simlin @aborjinik @DonkeyCo this bug is reported for the v2 Table - assigning you.

**Comment by DonkeyCo** [comment]
Regarding the scrollbar issue:
What is the exact expectation here?
The table has 5 columns with the following widths defined: 300px, auto, min. 300px, auto, 220px
The table has a total width specified of 900px.
Just already calculating the total width of columns that have a width specified, we end up with 820px. The columns that have no width specified have a width of 48px. We would end up with a total column width of 916px.
So even without selection, we'd need to overflow the table. That is to be expected.

Regarding the cut-off header text: #9754 is also mentioning this issue. We will discuss this internally, as SAPUI5 is behaving similarly.

**Comment by Lukas742** [comment]
Hi @DonkeyCo 

> What is the exact expectation here?
The table has 5 columns with the following widths defined: 300px, auto, min. 300px, auto, 220px
The table has a total width specified of 900px.
Just already calculating the total width of columns that have a width specified, we end up with 820px. The columns that have no width specified have a width of 48px. We would end up with a total column width of 916px.
So even without selection, we'd need to overflow the table. That is to be expected.

Sorry, my bad, you're totally right. I didn't realize that `min-width` was still applied 🤦

**Comment by DonkeyCo** [comment]
Sorry for the very late reply from my side. We've created an internal incident to track the cut-off header text issue DINC0298328

**Comment by DonkeyCo** [comment]
**TL;DR** I had a look at the truncation issue and this is by design. The default value of `minWidth` is `auto`, but if `minWidth` is set to `auto`, we ensure from our side, that the column cannot be smaller than 3rem.
If you want to circumvent truncation, you need to apply a fitting `minWidth`.

Some explanation to why we've decided to handle it like this:

The table internally uses a CSS Grid to layout it's header cells and cells. If we'd use `auto` in a CSS Grid, the width of the grid-column (e.g. `minmax(auto, [maxWidth])` would be as follows:
> As a minimum represents the largest minimum size of items in that track (specified by the [min-width](https://developer.mozilla.org/en-US/docs/Web/CSS/min-width)/[min-height](https://developer.mozilla.org/en-US/docs/Web/CSS/min-height) of the items). This is often, though not always, the [min-content](https://developer.mozilla.org/en-US/docs/Web/CSS/min-content) size.

This behaviour is fine for some cases, but runs into problems if the content could reach a width of 0px. I've created a [JSFiddle](https://jsfiddle.net/vgfantsd/), which shows the issue. As the first column will have a width of min-content, it's getting really small, as it is wrapping.
Therefore, to ensure that header cells and cells in general are clickable (and in future maybe even easily resizable), we apply this 3rem min-width.
We've got a similar behaviour in UI5 with the GridTable (sap.ui.table.Table).

I will enhance the documentation for the min-width property, specifically mentioning the 3rem restriction when using `auto` as value.
